### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by michaelogala**
TESTS
### WHAT DOES THIS PR DO?

This PR tests all the features that have been developed in this app. Making sure that all features behave according to design and requirements

### CODE REVIEW

File: app/controllers/redirects_controller.rb [hunk 1]
```diff
@@ -1,21 +1,18 @@
 class RedirectsController < ApplicationController
-include RedirectsHelper
-layout false
+  include RedirectsHelper
+  layout false
+
   def inactive
   end
```
Comment (mradeybee) [inline app/controllers/redirects_controller.rb:7]: What is the use of this method?

File: app/models/link.rb [hunk 2]
```diff
@@ -7,18 +7,18 @@ class Link < ActiveRecord::Base
   scope :popular, -> { order("links.clicks DESC").limit(5) }
   scope :find_url, -> { select('given_url').where(slug: :slug) }
 
-  after_create :generate_slug_and_scrape
+  # after_create :scrape_title
```
Comment (mradeybee) [inline app/models/link.rb:10]: Remove dead code.

File: coverage/.last_run.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "covered_percent": 95.12
+    "covered_percent": 98.67
```
Comment (mradeybee) [inline coverage/.last_run.json:3]: You dont need to commit this file and the whole coverage folder.

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -35,30 +56,33 @@
...
+      with.test_framework :rspec
 
-    with.library :rails
+      with.library :rails
+    end
   end
 end
+
+def sign_up
+  @password = Faker::Internet.password(8)
```
Comment (mradeybee) [inline spec/rails_helper.rb:74]: You can split this into multiple methods and call all of them in here.

**Review by mradeybee — CHANGES_REQUESTED** [review]

File: app/controllers/redirects_controller.rb [hunk 5]
```diff
@@ -1,21 +1,18 @@
...
-      p '*' * 70
-      puts link
-      if !link
-        render 'deleted'
-      elsif link.active?
-        visit_and_update(link)
-      else
-        render 'inactive'
-      end
+    link = Link.find_by(slug: params[:slug])
```
Comment (andela-fadebayo) [inline app/controllers/redirects_controller.rb:9]: This can be moved to Link model using `scope`.
`Link.find_by_slug(params[:slug])`

**Pull request merged**